AWSS3/AWSS3TransferUtility.m

```diff
@@ -2433,7 +2432,7 @@ - (void)URLSession:(NSURLSession *)session
     
         //Calculate the total sent so far
         int64_t totalSentSoFar = 0;
-        for (AWSS3TransferUtilityUploadSubTask *aSubTask in [transferUtilityMultiPartUploadTask.completedPartsDictionary allValues]) {
+        for (AWSS3TransferUtilityUploadSubTask *aSubTask in transferUtilityMultiPartUploadTask.completedPartsSet) {
```

Comment:
@lawmicha - This is the line of the patch that I still don't fully understand. If we are unconditionally adding `AWSS3TransferUtilityUploadSubTask` objects (of type `id`?) to `completedPartsSet` when we hydrate from the database, how do we not end up with duplicates here ini the accounting for the arithmetic in the for loop?

Reply:
good point, I'm not sure what is persisted in the database so there might be rows which have the same part numbers. This is more reason to have a dictionary based on partNumber. the defensive thing we can do here is similar to the logic in `callFinishMultiPartForUploadTask` by dedupping the completedPartsSet to a dictionary based on part numbers